src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exc.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkValidity(user);
+        storageDao.add(user);
+        return user;
     }
+
+    private void checkValidity(User user) {
+        if (user == null) {
+            throw new InvalidDataException("User is 'null'");
+        }
+        if (user.getAge() == null 
+                || user.getAge() < MIN_AGE) {
+            throw new InvalidDataException("Age equals 'null' or is less than 18");
+        }
+        if (user.getLogin() == null 
+                || user.getLogin().equals("")) {
+            throw new InvalidDataException("Login equals 'null' or is 0 characters long");
+        }
+        if (user.getPassword() == null
+                || user.getPassword().length() < 6) {
```

Comment:
[Checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation)
**Move all hardcoded values to [constant fields](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names) and give them informative names.**
You've done it with MIN_AGE but not with password length. Why?)